symbolic-base/src/ZkFold/Prelude.hs

```diff
@@ -32,6 +37,17 @@ iterateM :: Monad m => Natural -> (a -> m a) -> a -> m a
 iterateM 0 _ x = return x
 iterateM n f x = f x >>= iterateM (n - 1) f
 
+iterate :: (a -> a) -> a -> [a]
+iterate f x =
+    let x' = f x
+    in x' `seq` (x : iterate f x')
+
+iterateN :: (a -> a) -> Natural -> a -> [a]
```

Comment:
same here -- I think it would be better to name it `iterateN'` to indicate that it's a strict version